docs/design/design-patterns/non-functional-requirements-investigation-guide.md

```diff
@@ -0,0 +1,148 @@
+# Non-Functional Requirements Investigation
+
+## Background
+
+In software engineering, detailed investigation of non-functional requirements is often deferred until later stages of the project in favor of a more exclusive initial focus on the functional requirements.
```

Comment:
I would suggest rewording this section to not say what we _shouldn't do_ but actually what we _should do_, so cut out the cons and lead with the positives/benefits of adopting the practice. 

I also recommend making this section more clear and concise.

Reply:
Good points; done.